sw/host/opentitantool/src/command/emulator.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use erased_serde::Serialize;
+use std::any::Any;
+use std::collections::HashMap;
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+use opentitanlib::app::command::CommandDispatch;
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::io::emu::EmuValue;
+use opentitanlib::transport::Capability;
+
+#[derive(Debug, StructOpt)]
+/// Get State of Emulator instance
+pub struct EmuGetState {}
+
+#[derive(serde::Serialize)]
+pub struct EmuGetStateResult {
+    pub state: String,
```

Comment:
Why is this a string? stringly-typed things are typically a bad idea.

Reply:
Idea is to return something human readable. 
It is string generated from enum EmuState - It is safe in this context.
But as you have noticed, it may not be very optimal because EmuGetStateResult alredy derive Serializable.